src/GoTrueClient.ts

```diff
@@ -640,9 +564,13 @@ export default class GoTrueClient {
    * For server-side management, you can revoke all refresh tokens for a user by passing a user's JWT through to `auth.api.signOut(JWT: string)`. There is no way to revoke a user's session JWT before it automatically expires
    */
   async signOut(): Promise<{ error: AuthError | null }> {
-    const accessToken = this.currentSession?.access_token
+    const { session, error: sessionError } = await this.getSession()
+    if (sessionError) {
+      throw sessionError
+    }
+    const accessToken = session?.access_token
     this._removeSession()
-    this._notifyAllSubscribers('SIGNED_OUT')
+    this._notifyAllSubscribers('SIGNED_OUT', session)
```

Comment:
Could we pass `null` instead of `session`? So far `SIGNED_OUT` always emitted `null`. With this change `session` is most likely not `null` at this point.

Reply:
good point, thanks for catching this!